lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -37,29 +39,34 @@ def exists?
     script_content = ps_script_content('test')
     Puppet.debug "\n" + script_content
     output = powershell(powershell_args, script_content)
-    if ['true','false'].include?(output.to_s.strip.downcase)
-      check = (output.to_s.strip.downcase == 'true')
-      Puppet.debug "Dsc Resource Exists?: #{check}"
-      Puppet.debug "dsc_ensure: #{resource[:dsc_ensure]}" if resource.parameters.has_key?(:dsc_ensure)
-      Puppet.debug "ensure: #{resource[:ensure]}"
-      check
-    else
-      fail(output)
-    end
+    Puppet.debug "Dsc Resource returned: #{output}"
+    data = JSON.parse(output)
+    fail(data['errormessage']) if !data['errormessage'].empty?
+    check = (data['isdesiredstate'] == 'true')
```

Comment:
Oops.. one more typo, but I got you covered.

`isdesiredstate` should be `indesiredstate`